tesseract_python/tesseract_python/swig/tesseract_scene_graph/srdf_model.i

```diff
@@ -0,0 +1,113 @@
+/**
+ * @file srdf_parser.i
+ * @brief SWIG interface file for tesseract_scene_graph/parser/srdf_parser.h
+ *
+ * @author Hervé Audren
+ * @date January 20, 2020
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2020, Ascent Robotics Inc.
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+%feature(flatnested)
+
+%{
+#include <tesseract_scene_graph/srdf_model.h>
+%}
+
+%shared_ptr(tesseract_scene_graph::SRDFModel)
+
+namespace tesseract_scene_graph
+{
+
+class SRDFModel
+{
+public:
+  using Ptr = std::shared_ptr<SRDFModel>;
+  using ConstPtr = std::shared_ptr<const SRDFModel>;
+
+  /** @brief A structure to hold opw kinematics data */
+  struct OPWKinematicParameters
+  {
+    double a1, a2, b, c1, c2, c3, c4;
+    double offsets[6];
+    signed char sign_corrections[6];
+  };
+
+  using JointState = std::unordered_map<std::string, double>;
+  using JointStates = std::unordered_map<std::string, JointState>;
+  using GroupStates = std::unordered_map<std::string, JointStates>;
+  using TCPs = std::unordered_map<std::string, Eigen::Isometry3d>;
+  using GroupTCPs = std::unordered_map<std::string, TCPs>;
+  using ChainGroups = std::unordered_map<std::string, std::vector<std::pair<std::string, std::string>>>;
+  using JointGroups = std::unordered_map<std::string, std::vector<std::string>>;
+  using LinkGroups = std::unordered_map<std::string, std::vector<std::string>>;
+  using GroupOPWKinematics = std::unordered_map<std::string, OPWKinematicParameters>;
+
+  /** @brief Load Model from TiXMLElement */
+  bool initXml(const tesseract_scene_graph::SceneGraph& scene_graph, tinyxml2::XMLElement* srdf_xml);
+
+  /** @brief Load Model from TiXMLDocument */
+  bool initXml(const tesseract_scene_graph::SceneGraph& scene_graph, tinyxml2::XMLDocument* srdf_xml);
+
+  /** @brief Load Model given a filename */
+  bool initFile(const tesseract_scene_graph::SceneGraph& scene_graph, const std::string& filename);
+
+  /** @brief Load Model from a XML-string */
+  bool initString(const tesseract_scene_graph::SceneGraph& scene_graph, const std::string& xmlstring);
+
+  /** @brief Save the model to a file */
+  bool saveToFile(const std::string& file_path) const;
+
+  /** @brief Get the name of this model */
+  const std::string& getName() const;
```

Comment:
Don't return string references, they are not correctly handled by SWIG. You can safely return std::string instead.

Reply:
Should I just wrap it and return a copy then on the python side?

Reply:
If you set the return type to `std::string` in the SWIG *.i file it will create a copy and behave properly. I have fixed these in my pull request.